linode/networkipassignment/framework_resource.go

```diff
@@ -0,0 +1,154 @@
+package networkipassignment
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_networking_assign_ip",
```

Comment:
Could this resource be renamed to `linode_networking_ip_assignment` to match the declarative naming scheme?

Reply:
Yeah I agree that makes more sense. I will change the resource name too.